R/mod_read_spreadsheet.R

```diff
@@ -33,35 +59,75 @@ mod_read_spreadsheet_ui <- function(id){
 #' @keywords internal
     
 mod_read_spreadsheet_server <- function(id) {
-  # File uploading limit: 9MB.
+  # File uploading limit: 9MB
   options(shiny.maxRequestSize = 9*1024^2)
   
   moduleServer(id, function(input, output, session) {
-    # Reading infosheet from local
-    table_data <- eventReactive(input$file, {
-      # File input requirement
-      req(input$file)
+    # Create one activate reactive from two buttons
+    # If either of the buttons are pressed the reactive fires
+    activate <- reactive(
+      if (input$upload_url == 0 & input$upload_file == 0) {
+        NULL
+      } else {
+        input$upload_url + input$upload_file
+      }
+    )
+    
+    # Decide which input to read
+    # The app reads the input of the latest upload button is pressed
+    which_input <- reactiveVal(NULL)
+    
+    observe({
+      req(input$upload_file)
+      which_input("file")
+    })
+    
+    observe({
+      req(input$upload_url)
+      which_input("url")
+    })
+    
+    # Reading infosheet
+    table_data <- eventReactive(activate(), {
+      # Googlesheets authentication
+      googlesheets4::gs4_deauth()
       
-      read_infosheet(infosheet_path = input$file$datapath)
+      # Select the path to read based on which action button is pressed
+      if (which_input() == "file") {
+        if (is.null(input$file)) {
+          infosheet_path <- ""
+          } else {
+            infosheet_path <- input$file$datapath
+            }
+        } else if (which_input() == "url") {
+          infosheet_path <- input$url
+          }
+      
+      # Reading data
+      read_infosheet(infosheet_path = infosheet_path)
       })
     
     # Alert modal if infosheet is incomplete
-    valid_infosheet <- mod_check_modal_server("check_modal_ui_1", activate = reactive(input$file), table_data = table_data)
-    
+    valid_infosheet <- mod_check_modal_server("check_modal_ui_1", activate = activate, table_data = table_data)
+
     # Delete empty rows
-    table_data_clean <- eventReactive(valid_infosheet, {
+    table_data_clean <- eventReactive(activate(), {
+      if (valid_infosheet() == TRUE) {
       table_data() %>%
       tibble::as_tibble() %>%
       dplyr::filter_at(
         dplyr::vars(Firstname, `Middle name`, Surname),
         dplyr::any_vars(!is.na(.)))
+      } else {
+        NULL
```

Comment:
maybe we should return something here, so one can see the sheet and why it was not validated ?

```suggestion
        table_data() %>%
        tibble::as_tibble() %>%
        tibble::add_row("Firstname"="warning: this infosheet did not pass validation, please see the errors that appened during validation, .before = 0)
```

Reply:
The validation code is still running and upon loading an infosheet the modal pop up with the result of the validation that explains why it was not validated.

The reason why we do not show the sheet until it throws an error is that there is a custom code that highlights the row with missing credit info which needs the credit columns to be present in the infosheet currently.

We can change that behavior but I think that should be in a different pull request not in this one.

Reply:
Then please create a new PR with this code, otherwise we will forget about this.

Reply:
new issue created #51